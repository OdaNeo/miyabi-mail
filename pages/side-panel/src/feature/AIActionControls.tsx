import { Button, Progress, Select, SelectContent, SelectItem, SelectTrigger, SelectValue } from '@extension/ui';
import { Languages, Wand2Icon } from 'lucide-react';
import { useI18n } from '../hooks/useI18n';
import { rolePrompt } from '../utils/tts';
import { useProgress } from '../hooks/useProgress';
import { useGeneratedStore } from '../store/generatedStore';
import { useExpandedSectionStore } from '../store/expandedSectionStore';
import {
  apiVersionStorage,
  languageStorage,
  replyStorage,
  temperatureStorage,
  translationStorage,
} from '@extension/storage';
import { useReplyLoading } from '@/store/replyLoadingStore';
import { useTranslationLoading } from '@/store/translationLoadingStore';
import { usePolishingLoading } from '@/store/polishingLoadingStore';
import { useStorage } from '@extension/shared';
import { LANGUAGES } from '@/const/language';
import { addTaskRecord } from '@/db/taskRecordsRepository';
import { TaskHistoryMapper } from '@/db/Dto';

export function AIActionControls({
  inputText,
  subject,
  setInputText,
  runOpenAIAction,
  handleOnClick,
}: {
  inputText: string;
  subject: string;
  setInputText: (text: string) => void;
  runOpenAIAction: ({
    prompt,
    onStart,
    onFinish,
    onSuccess,
  }: {
    prompt: string;
    onStart?: () => void;
    onFinish?: () => void;
    onSuccess?: (content: string) => void;
  }) => Promise<void>;
  handleOnClick: () => void;
}) {
  const { IS_POLISHING, POLISH, IS_TRANSLATING, TRANSLATION, IS_GENERATING, GENERATE } = useI18n();

  const language = useStorage(languageStorage);

  const { setIsGenerated } = useGeneratedStore();
  const { setExpandedSection } = useExpandedSectionStore();
  const { isReplyLoading, setIsReplyLoading } = useReplyLoading();
  const { isTranslationLoading, setIsTranslationLoading } = useTranslationLoading();
  const { isPolishingLoading, setIsPolishingLoading } = usePolishingLoading();

  const [replyProgress, setReplyProgress] = useProgress(isReplyLoading);
  const [translationProgress, setTranslationProgress] = useProgress(isTranslationLoading);
  const [polishingProgress, setPolishingProgress] = useProgress(isPolishingLoading);

  const isLoading = isReplyLoading || isTranslationLoading || isPolishingLoading;

  const apiVersion = useStorage(apiVersionStorage);
  const temperature = useStorage(temperatureStorage);

  const handleReply = () => {
    const createTime = new Date().toISOString();
    runOpenAIAction({
      prompt: rolePrompt(inputText?.trim(), 'REPLY', language, subject?.trim()),
      onStart: () => {
        setIsGenerated(false), setIsReplyLoading(true), setReplyProgress(0);
      },
      onFinish: () => {
        setIsReplyLoading(false), setReplyProgress(0);
      },
      onSuccess: content => {
        setIsGenerated(true), replyStorage.set(content), setExpandedSection('REPLY');
        addTaskRecord(
          TaskHistoryMapper.toEntity({
            inputText,
            language,
            promptKey: 'REPLY',
            apiVersion,
            temperature,
            subject,
            result: content,
            createTime,
            completedTime: new Date().toISOString(),
          }),
        );
      },
    });
  };

  const handleTranslation = () => {
    const createTime = new Date().toISOString();
    runOpenAIAction({
      prompt: rolePrompt(inputText?.trim(), 'TRANSLATION', language),
      onStart: () => {
        setIsGenerated(false), setIsTranslationLoading(true), setTranslationProgress(0);
      },
      onFinish: () => {
        setIsTranslationLoading(false), setTranslationProgress(0);
      },
      onSuccess: content => {
        setIsGenerated(true), translationStorage.set(content), setExpandedSection('TRANSLATION');
        addTaskRecord(
          TaskHistoryMapper.toEntity({
            inputText,
            language,
            promptKey: 'TRANSLATION',
            apiVersion,
            temperature,
            result: content,
            createTime,
            completedTime: new Date().toISOString(),
          }),
        );
      },
    });
  };

  const handlePolishing = () => {
    const createTime = new Date().toISOString();
    runOpenAIAction({
      prompt: rolePrompt(inputText?.trim(), 'POLISHING'),
      onStart: () => {
        setIsGenerated(false), setIsPolishingLoading(true), setPolishingProgress(0);
      },
      onFinish: () => {
        setIsPolishingLoading(false), setPolishingProgress(0);
      },
      onSuccess: content => {
        setIsGenerated(false), setInputText(content), setExpandedSection(null);
        addTaskRecord(
          TaskHistoryMapper.toEntity({
            inputText,
            language,
            promptKey: 'POLISHING',
            apiVersion,
            temperature,
            result: content,
            createTime,
            completedTime: new Date().toISOString(),
          }),
        );
      },
    });
  };

  return (
    <div
      className="flex items-center px-5 py-3 border-t border-[#e5e7eb] dark:border-[#2e3238] bg-[#f3f4f6] dark:bg-[#23252b]"
      onClick={handleOnClick}
    >
      <Select disabled={isLoading} value={language} onValueChange={languageStorage.set}>
        <SelectTrigger className="w-26 h-9 rounded-md border-[#e5e7eb] dark:border-[#2e3238] bg-white dark:bg-[#1f2128]">
          <SelectValue placeholder="选择语言" />
        </SelectTrigger>
        <SelectContent>
          {LANGUAGES.map(lang => (
            <SelectItem key={lang.value} value={lang.value} className="text-sm">
              {lang.label}
            </SelectItem>
          ))}
        </SelectContent>
      </Select>
      <div className="flex-1" />
      <Button
        onClick={handlePolishing}
        disabled={isLoading || !inputText?.trim()}
        className="relative ml-2 h-9 px-3 text-ms
          bg-slate-200 dark:bg-slate-700 hover:bg-[#f9fafb] dark:hover:bg-[#2e3238]"
      >
        <Wand2Icon className="h-4 w-4 text-[#8b5cf6] dark:text-[#a78bfa] opacity-80" />
        {isPolishingLoading ? <span>{IS_POLISHING}</span> : <span>{POLISH}</span>}
        {isPolishingLoading && (
          <Progress
            value={polishingProgress}
            className="absolute bottom-0 left-0 right-0 h-1 bg-[#e5e7eb] dark:bg-[#2e3238]"
            indicatorClassName="bg-[#8b5cf6] dark:bg-[#a78bfa]"
          />
        )}
      </Button>
      <Button
        onClick={handleTranslation}
        disabled={isLoading || !inputText?.trim()}
        className="relative ml-2 h-9 px-3 text-ms
          bg-slate-200 dark:bg-slate-700 hover:bg-[#f9fafb] dark:hover:bg-[#2e3238]"
      >
        <Languages className="h-4 w-4 text-[#8b5cf6] dark:text-[#a78bfa] opacity-80" />
        {isTranslationLoading ? <span>{IS_TRANSLATING}</span> : <span>{TRANSLATION}</span>}
        {isTranslationLoading && (
          <Progress
            value={translationProgress}
            className="absolute bottom-0 left-0 right-0 h-1 bg-[#e5e7eb] dark:bg-[#2e3238]"
            indicatorClassName="bg-[#8b5cf6] dark:bg-[#a78bfa]"
          />
        )}
      </Button>
      <Button
        onClick={handleReply}
        data-testid="reply-button"
        disabled={isLoading || !inputText?.trim()}
        className="relative overflow-hidden ml-2 h-9 px-3 text-ms text-white font-normal
          rounded-md shadow-sm hover:shadow
          bg-[#8b5cf6] hover:bg-[#7c3aed] dark:bg-[#a78bfa] dark:hover:bg-[#9061f9]"
      >
        {isReplyLoading ? <span>{IS_GENERATING}</span> : <span>{GENERATE}</span>}
        {isReplyLoading && (
          <Progress
            value={replyProgress}
            className="absolute bottom-0 left-0 right-0 h-1 bg-[#8b5cf6] dark:bg-[#a78bfa]"
            indicatorClassName="bg-[#e5e7eb] dark:bg-[#2e3238]"
          />
        )}
      </Button>
    </div>
  );
}
