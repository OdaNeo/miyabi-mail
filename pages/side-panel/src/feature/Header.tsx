import {
  Button,
  Tooltip,
  TooltipContent,
  TooltipProvider,
  TooltipTrigger,
  Popover,
  PopoverContent,
  PopoverTrigger,
} from '@extension/ui';
import { Settings, Sun, Moon, Globe, Feather, Clock } from 'lucide-react';
import { useStorage } from '@extension/shared';
import { darkModeStorage, i18nStorage } from '@extension/storage';
import { useOpenStore } from '@/store/openStore';
import { useI18n } from '@/hooks/useI18n';
import { SettingPopoverContent } from './SettingPopoverContent';
import { useHistoryOpenStore } from '@/store/historyOpenStore';

export function Header() {
  const darkMode = useStorage(darkModeStorage);

  const { toggleIsHistoryOpen } = useHistoryOpenStore();

  const { isOpen, setIsOpen } = useOpenStore();

  const { CHANGE_TO_LIGHT, CHANGE_TO_DARK, CHANGE_LANGUAGE } = useI18n();

  const handleSwitchLanguage = () => {
    i18nStorage.next();
  };

  const handleToggleDarkMode = () => {
    document.documentElement.classList.toggle('dark');
    darkModeStorage.set(!darkMode);
  };

  const handleTogglePopover = () => {
    setIsOpen(!isOpen);
  };

  return (
    <div className="flex justify-between items-center px-5 py-3 bg-gradient-to-r from-[#e9ecf1] to-[#f0f2f5] dark:from-[#23252b] dark:to-[#282a32] border-b border-[#e5e7eb] dark:border-[#2e3238]">
      <Feather className="w-5 h-5 text-[#8b5cf6] dark:text-[#a78bfa] opacity-80" />
      <h1 className="text-xl font-medium tracking-widest flex items-center ml-2">雅返信</h1>
      <div className="flex-1" />
      <div className="flex space-x-2">
        <Button
          variant="ghost"
          size="icon"
          className="w-8 h-8 text-slate-600 dark:text-slate-400 hover:bg-[#e2e3e4] dark:hover:bg-[#2e3238]"
          onClick={() => {
            toggleIsHistoryOpen();
          }}
        >
          <Clock data-testid="clock-icon" />
        </Button>
        <TooltipProvider>
          <Tooltip>
            <TooltipTrigger asChild>
              <Button
                variant="ghost"
                size="icon"
                className="w-8 h-8 text-slate-600 dark:text-slate-400 hover:bg-[#e2e3e4] dark:hover:bg-[#2e3238]"
                onClick={handleSwitchLanguage}
              >
                <Globe data-testid="globe-icon" />
              </Button>
            </TooltipTrigger>
            <TooltipContent className="bg-slate-700 text-slate-100 dark:bg-slate-200 dark:text-slate-800">
              {CHANGE_LANGUAGE}
            </TooltipContent>
          </Tooltip>
        </TooltipProvider>
        <TooltipProvider>
          <Tooltip>
            <TooltipTrigger asChild>
              <Button
                variant="ghost"
                size="icon"
                data-testid="toggle-dark-mode"
                className="w-8 h-8 text-slate-600 dark:text-slate-400 hover:bg-[#e2e3e4] dark:hover:bg-[#2e3238]"
                onClick={handleToggleDarkMode}
              >
                {darkMode ? <Sun className="h-4 w-4" /> : <Moon className="h-4 w-4" />}
              </Button>
            </TooltipTrigger>
            <TooltipContent className="bg-slate-700 text-slate-100 dark:bg-slate-200 dark:text-slate-800">
              <p>{darkMode ? CHANGE_TO_LIGHT : CHANGE_TO_DARK}</p>
            </TooltipContent>
          </Tooltip>
        </TooltipProvider>
        <Popover open={isOpen} onOpenChange={setIsOpen}>
          <PopoverTrigger asChild>
            <Button
              onClick={handleTogglePopover}
              variant="ghost"
              size="icon"
              data-testid="setting-icon"
              className="w-8 h-8 text-slate-600 dark:text-slate-400 hover:bg-[#e2e3e4] dark:hover:bg-[#2e3238]"
            >
              <Settings className="h-4 w-4" />
            </Button>
          </PopoverTrigger>
          <PopoverContent className="w-60 rounded-lg bg-white dark:bg-slate-900 border border-slate-200 dark:border-slate-700">
            <SettingPopoverContent setIsOpen={setIsOpen} />
          </PopoverContent>
        </Popover>
      </div>
    </div>
  );
}
