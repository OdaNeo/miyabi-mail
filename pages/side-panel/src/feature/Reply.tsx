import type { PROMPT_KEYS } from '@/utils/tts';
import { AnimatePresence, motion } from 'framer-motion';
import { Label } from '@extension/ui';
import { ChevronDown, ChevronUp } from 'lucide-react';
import { MessageBox } from '../components/MessageBox';
import { replyStorage, translationStorage } from '@extension/storage';
import { useStorage } from '@extension/shared';
import { useGeneratedStore } from '@/store/generatedStore';
import { useExpandedSectionStore } from '@/store/expandedSectionStore';
import { useI18n } from '@/hooks/useI18n';
import { CopyButton } from '@/components/CopyButton';

export function Reply() {
  const reply = useStorage(replyStorage);
  const translation = useStorage(translationStorage);

  const { isGenerated } = useGeneratedStore();
  const { expandedSection, setExpandedSection } = useExpandedSectionStore();

  const { ORIGINAL_TRANSLATION, REPLY } = useI18n();

  const toggleSection = (section: PROMPT_KEYS) => {
    setExpandedSection(expandedSection === section ? null : section);
  };

  const handleCopy = (section: PROMPT_KEYS) => {
    navigator.clipboard.writeText(section === 'REPLY' ? reply : translation);
  };

  return (
    <AnimatePresence>
      {isGenerated && (
        <motion.div
          initial={{ opacity: 0, height: 0 }}
          animate={{ opacity: 1, height: 'auto' }}
          exit={{ opacity: 0, height: 0 }}
          className="overflow-y-auto scrollbar-custom rounded-lg border border-[#e5e7eb] dark:border-[#2e3238]
        dark:bg-slate-800 dark:text-slate-100 dark:placeholder-slate-400 bg-white text-slate-900 placeholder-slate-500"
        >
          {(['TRANSLATION', 'REPLY'] as PROMPT_KEYS[]).map(section => (
            <div key={section}>
              <div
                className="flex justify-between items-center px-3 h-9 cursor-pointer bg-slate-100 dark:bg-[#23252b]
                 hover:bg-slate-200 dark:hover:bg-slate-800"
                onClick={() => toggleSection(section)}
                onKeyUp={e => {
                  if (e.key === 'Enter' || e.key === ' ') {
                    toggleSection(section);
                  }
                }}
                data-testid={`section-${section.toLocaleLowerCase()}`}
                role="button"
                tabIndex={0}
              >
                <Label className="font-medium text-[#4b5563] dark:text-[#d1d5db]">
                  {section === 'TRANSLATION' ? ORIGINAL_TRANSLATION : REPLY}
                </Label>
                <div className="flex items-center space-x-2">
                  <CopyButton handleCopyText={() => handleCopy(section)} />
                  {expandedSection === section ? (
                    <ChevronUp className="h-3 w-3 text-slate-500 dark:text-slate-400" />
                  ) : (
                    <ChevronDown className="h-3 w-3 text-slate-500 dark:text-slate-400" />
                  )}
                </div>
              </div>
              {expandedSection === section && (
                <MessageBox>{section === 'TRANSLATION' ? translation : reply}</MessageBox>
              )}
            </div>
          ))}
        </motion.div>
      )}
    </AnimatePresence>
  );
}
