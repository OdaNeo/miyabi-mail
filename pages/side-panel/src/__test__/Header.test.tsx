import { render, fireEvent, screen } from '@testing-library/react';
import { describe, it, expect, vi, beforeEach } from 'vitest';
import { Header } from '../feature/Header';
import { useStorage } from '@extension/shared';
import { i18nStorage } from '@extension/storage';

vi.mock('@extension/shared', () => ({
  useStorage: vi.fn(),
}));

vi.mock('@extension/storage', () => ({
  darkModeStorage: {
    set: vi.fn(),
    next: vi.fn(),
  },
  i18nStorage: {
    next: vi.fn(),
  },
}));

const setIsOpenMock = vi.fn();

vi.mock('@/store/openStore', () => ({
  useOpenStore: () => ({
    isOpen: false,
    setIsOpen: setIsOpenMock,
  }),
}));

const mockToggleIsHistoryOpen = vi.fn();
vi.mock('@/store/historyOpenStore', () => ({
  useHistoryOpenStore: () => ({
    isHistoryOpen: false,
    toggleIsHistoryOpen: mockToggleIsHistoryOpen,
  }),
}));

vi.mock('@/hooks/useI18n', () => ({
  useI18n: () => ({
    CHANGE_TO_LIGHT: 'Change to Light',
    CHANGE_TO_DARK: 'Change to Dark',
    CHANGE_LANGUAGE: 'Change Language',
  }),
}));

describe('Header Component', () => {
  const useStorageMock = vi.mocked(useStorage);

  beforeEach(() => {
    vi.clearAllMocks();
    document.documentElement.classList.remove('dark');
  });

  it('should render correctly', () => {
    useStorageMock.mockReturnValue(false);
    const { container } = render(<Header />);
    expect(container).toMatchSnapshot();
  });

  it('should render correctly in dark mode', () => {
    useStorageMock.mockReturnValue(true);
    const { container } = render(<Header />);
    expect(container).toMatchSnapshot();
  });

  it('should toggle dark mode', () => {
    useStorageMock.mockReturnValue(false);
    render(<Header />);
    const toggleButton = screen.getByTestId('toggle-dark-mode');
    fireEvent.click(toggleButton);
    expect(document.documentElement.classList.contains('dark')).toBe(true);
  });

  it('should handle language switch', () => {
    useStorageMock.mockReturnValue(false);
    render(<Header />);
    const languageButton = screen.getByTestId('globe-icon').parentElement;
    fireEvent.click(languageButton!);
    expect(vi.mocked(i18nStorage.next)).toHaveBeenCalled();
  });

  it('should handle settings popover', () => {
    useStorageMock.mockReturnValue(false);
    render(<Header />);
    const settingsButton = screen.getByTestId('setting-icon');
    fireEvent.click(settingsButton);
    expect(setIsOpenMock).toHaveBeenCalledWith(true);
  });

  it('should show history when clicking clock icon', () => {
    render(<Header />);
    const clockButton = screen.getByTestId('clock-icon');
    fireEvent.click(clockButton);
    expect(vi.mocked(mockToggleIsHistoryOpen)).toHaveBeenCalled();
  });
});
