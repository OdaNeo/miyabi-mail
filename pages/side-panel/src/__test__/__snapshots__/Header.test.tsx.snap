// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header Component > should render correctly 1`] = `
<div>
  <div
    class="flex justify-between items-center px-5 py-3 bg-gradient-to-r from-[#e9ecf1] to-[#f0f2f5] dark:from-[#23252b] dark:to-[#282a32] border-b border-[#e5e7eb] dark:border-[#2e3238]"
  >
    <svg
      class="lucide lucide-feather w-5 h-5 text-[#8b5cf6] dark:text-[#a78bfa] opacity-80"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12.67 19a2 2 0 0 0 1.416-.588l6.154-6.172a6 6 0 0 0-8.49-8.49L5.586 9.914A2 2 0 0 0 5 11.328V18a1 1 0 0 0 1 1z"
      />
      <path
        d="M16 8 2 22"
      />
      <path
        d="M17.5 15H9"
      />
    </svg>
    <h1
      class="text-xl font-medium tracking-widest flex items-center ml-2"
    >
      雅返信
    </h1>
    <div
      class="flex-1"
    />
    <div
      class="flex space-x-2"
    >
      <button
        class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 hover:text-accent-foreground w-8 h-8 text-slate-600 dark:text-slate-400 hover:bg-[#e2e3e4] dark:hover:bg-[#2e3238]"
      >
        <svg
          class="lucide lucide-clock"
          data-testid="clock-icon"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="12"
            cy="12"
            r="10"
          />
          <polyline
            points="12 6 12 12 16 14"
          />
        </svg>
      </button>
      <button
        class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 hover:text-accent-foreground w-8 h-8 text-slate-600 dark:text-slate-400 hover:bg-[#e2e3e4] dark:hover:bg-[#2e3238]"
        data-state="closed"
      >
        <svg
          class="lucide lucide-globe"
          data-testid="globe-icon"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="12"
            cy="12"
            r="10"
          />
          <path
            d="M12 2a14.5 14.5 0 0 0 0 20 14.5 14.5 0 0 0 0-20"
          />
          <path
            d="M2 12h20"
          />
        </svg>
      </button>
      <button
        class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 hover:text-accent-foreground w-8 h-8 text-slate-600 dark:text-slate-400 hover:bg-[#e2e3e4] dark:hover:bg-[#2e3238]"
        data-state="closed"
        data-testid="toggle-dark-mode"
      >
        <svg
          class="lucide lucide-moon h-4 w-4"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 3a6 6 0 0 0 9 9 9 9 0 1 1-9-9Z"
          />
        </svg>
      </button>
      <button
        aria-controls="radix-:r2:"
        aria-expanded="false"
        aria-haspopup="dialog"
        class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 hover:text-accent-foreground w-8 h-8 text-slate-600 dark:text-slate-400 hover:bg-[#e2e3e4] dark:hover:bg-[#2e3238]"
        data-state="closed"
        data-testid="setting-icon"
        type="button"
      >
        <svg
          class="lucide lucide-settings h-4 w-4"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12.22 2h-.44a2 2 0 0 0-2 2v.18a2 2 0 0 1-1 1.73l-.43.25a2 2 0 0 1-2 0l-.15-.08a2 2 0 0 0-2.73.73l-.22.38a2 2 0 0 0 .73 2.73l.15.1a2 2 0 0 1 1 1.72v.51a2 2 0 0 1-1 1.74l-.15.09a2 2 0 0 0-.73 2.73l.22.38a2 2 0 0 0 2.73.73l.15-.08a2 2 0 0 1 2 0l.43.25a2 2 0 0 1 1 1.73V20a2 2 0 0 0 2 2h.44a2 2 0 0 0 2-2v-.18a2 2 0 0 1 1-1.73l.43-.25a2 2 0 0 1 2 0l.15.08a2 2 0 0 0 2.73-.73l.22-.39a2 2 0 0 0-.73-2.73l-.15-.08a2 2 0 0 1-1-1.74v-.5a2 2 0 0 1 1-1.74l.15-.09a2 2 0 0 0 .73-2.73l-.22-.38a2 2 0 0 0-2.73-.73l-.15.08a2 2 0 0 1-2 0l-.43-.25a2 2 0 0 1-1-1.73V4a2 2 0 0 0-2-2z"
          />
          <circle
            cx="12"
            cy="12"
            r="3"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;

exports[`Header Component > should render correctly in dark mode 1`] = `
<div>
  <div
    class="flex justify-between items-center px-5 py-3 bg-gradient-to-r from-[#e9ecf1] to-[#f0f2f5] dark:from-[#23252b] dark:to-[#282a32] border-b border-[#e5e7eb] dark:border-[#2e3238]"
  >
    <svg
      class="lucide lucide-feather w-5 h-5 text-[#8b5cf6] dark:text-[#a78bfa] opacity-80"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12.67 19a2 2 0 0 0 1.416-.588l6.154-6.172a6 6 0 0 0-8.49-8.49L5.586 9.914A2 2 0 0 0 5 11.328V18a1 1 0 0 0 1 1z"
      />
      <path
        d="M16 8 2 22"
      />
      <path
        d="M17.5 15H9"
      />
    </svg>
    <h1
      class="text-xl font-medium tracking-widest flex items-center ml-2"
    >
      雅返信
    </h1>
    <div
      class="flex-1"
    />
    <div
      class="flex space-x-2"
    >
      <button
        class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 hover:text-accent-foreground w-8 h-8 text-slate-600 dark:text-slate-400 hover:bg-[#e2e3e4] dark:hover:bg-[#2e3238]"
      >
        <svg
          class="lucide lucide-clock"
          data-testid="clock-icon"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="12"
            cy="12"
            r="10"
          />
          <polyline
            points="12 6 12 12 16 14"
          />
        </svg>
      </button>
      <button
        class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 hover:text-accent-foreground w-8 h-8 text-slate-600 dark:text-slate-400 hover:bg-[#e2e3e4] dark:hover:bg-[#2e3238]"
        data-state="closed"
      >
        <svg
          class="lucide lucide-globe"
          data-testid="globe-icon"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="12"
            cy="12"
            r="10"
          />
          <path
            d="M12 2a14.5 14.5 0 0 0 0 20 14.5 14.5 0 0 0 0-20"
          />
          <path
            d="M2 12h20"
          />
        </svg>
      </button>
      <button
        class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 hover:text-accent-foreground w-8 h-8 text-slate-600 dark:text-slate-400 hover:bg-[#e2e3e4] dark:hover:bg-[#2e3238]"
        data-state="closed"
        data-testid="toggle-dark-mode"
      >
        <svg
          class="lucide lucide-sun h-4 w-4"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="12"
            cy="12"
            r="4"
          />
          <path
            d="M12 2v2"
          />
          <path
            d="M12 20v2"
          />
          <path
            d="m4.93 4.93 1.41 1.41"
          />
          <path
            d="m17.66 17.66 1.41 1.41"
          />
          <path
            d="M2 12h2"
          />
          <path
            d="M20 12h2"
          />
          <path
            d="m6.34 17.66-1.41 1.41"
          />
          <path
            d="m19.07 4.93-1.41 1.41"
          />
        </svg>
      </button>
      <button
        aria-controls="radix-:r5:"
        aria-expanded="false"
        aria-haspopup="dialog"
        class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 hover:text-accent-foreground w-8 h-8 text-slate-600 dark:text-slate-400 hover:bg-[#e2e3e4] dark:hover:bg-[#2e3238]"
        data-state="closed"
        data-testid="setting-icon"
        type="button"
      >
        <svg
          class="lucide lucide-settings h-4 w-4"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12.22 2h-.44a2 2 0 0 0-2 2v.18a2 2 0 0 1-1 1.73l-.43.25a2 2 0 0 1-2 0l-.15-.08a2 2 0 0 0-2.73.73l-.22.38a2 2 0 0 0 .73 2.73l.15.1a2 2 0 0 1 1 1.72v.51a2 2 0 0 1-1 1.74l-.15.09a2 2 0 0 0-.73 2.73l.22.38a2 2 0 0 0 2.73.73l.15-.08a2 2 0 0 1 2 0l.43.25a2 2 0 0 1 1 1.73V20a2 2 0 0 0 2 2h.44a2 2 0 0 0 2-2v-.18a2 2 0 0 1 1-1.73l.43-.25a2 2 0 0 1 2 0l.15.08a2 2 0 0 0 2.73-.73l.22-.39a2 2 0 0 0-.73-2.73l-.15-.08a2 2 0 0 1-1-1.74v-.5a2 2 0 0 1 1-1.74l.15-.09a2 2 0 0 0 .73-2.73l-.22-.38a2 2 0 0 0-2.73-.73l-.15.08a2 2 0 0 1-2 0l-.43-.25a2 2 0 0 1-1-1.73V4a2 2 0 0 0-2-2z"
          />
          <circle
            cx="12"
            cy="12"
            r="3"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;
